### PR TITLE
Support running multiple media repos.

### PR DESCRIPTION
**Pull request by erikjohnston**
This requires a new config option to specify which media repo should be responsible for running background jobs to e.g. clear out expired URL preview caches. 

This is a bit of an icky way of doing it, though I think somewhat consistent with how we do worker configs currently. There are a couple of other options here:

1. Use distributed lock based on Redis, this is a bit difficult due to redis not having native support for this (the documentation suggests using a thing called [Redlock](https://redis.io/topics/distlock), but the fact that [Kleppman](http://martin.kleppmann.com/2016/02/08/how-to-do-distributed-locking.html) thinks its unsound is not exactly confidence inspiring). An argument could be made that we're probably will need to do something like this at some point though.
2. Change the way we do our config to instead list which instances are responsible for media operations in the main config and chose the first to be responsible for running the background jobs. This has the advantage that a) its impossible to mess up and b) its similar to how we specify which worker is responsible for event persistence.

TBH though, it feels like we shouldn't block this behind trying to implement a distributed lock or changing how we configure workers

### CODE REVIEW

**Comment by clokep** [comment]
I think the implementation looks sane, I don't have an opinion about using a config option vs. other approaches.

**Comment by clokep** [comment]
Removing the review flag on this as Rich asked for some changes in #synapse-dev:matrix.org.

**Comment by erikjohnston** [comment]
Ok, this now uses a scheme where you specify which instance runs the background jobs. Unfortunately, to support backwards compat we can't require it when using split out media repo and so if you don't specify the config option it'll still run on all of them.

File: docs/workers.md [hunk 1]
```diff
@@ -307,7 +307,12 @@ expose the `media` resource. For example:
            - media
 ```
 
-Note this worker cannot be load-balanced: only one instance should be active.
+Note that if running multiple media repositories they must be on the same server
+and you must configure which single instance to run the background tasks, e.g.:
```
Comment (clokep) [inline docs/workers.md:311]: I'm finding this sentence a bit awkward to read, maybe the following is clearer?

```suggestion
Note that if running multiple media repositories they must be on the same server
and you must configure a single instance to run the background tasks, e.g.:
```

**Pull request merged**